### PR TITLE
mgr/dashboard: Allow the decrease of pg's of an existing pool

### PR DESCRIPTION
**Pull request by Devp00l**
Now it's possible to decrease the pg's of an existing pool. The decrease
will take place over time like the increase. As for both no notification
will be created when this is done (issue #39482).

Fixes: https://tracker.ceph.com/issues/35917
Signed-off-by: Stephan Müller <smueller@suse.com>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.spec.ts [hunk 1]
```diff
@@ -1050,9 +1045,13 @@ describe('PoolFormComponent', () => {
         expect(form.getValue('ratio')).toBe(pool.options.compression_required_ratio);
       });
 
-      it('is only be possible to use the same or more pgs like before', () => {
+      it('should pg update should work on the first change not the second', () => {
+        testPgUpdate(undefined, -1, 16);
+      });
+
+      it('is possible to use the less or more pgs like before', () => {
```
Comment (LenzGr) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.spec.ts]:
```suggestion
      it('is possible to use the less or more pgs than before', () => {
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.spec.ts [hunk 2]
```diff
@@ -1050,9 +1045,13 @@ describe('PoolFormComponent', () => {
         expect(form.getValue('ratio')).toBe(pool.options.compression_required_ratio);
       });
 
-      it('is only be possible to use the same or more pgs like before', () => {
+      it('should pg update should work on the first change not the second', () => {
```
Comment (s0nea) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.spec.ts]: Is the first 'should' on purpose?

```suggestion
      it('pg update should work on the first change not the second', () => {
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.spec.ts (hunk 1 above)
Comment (s0nea) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.spec.ts]: Maybe?

```suggestion
      it('it should be possible to use more or fewer pgs than before', () => {
```

**Comment by Devp00l** [comment]
Addressed all comments

**Comment by callithea** [comment]
Tested it on my local system, works for me! The only thing I was wondering about is if the background task really should be under "finished" when the PGs are still being decreased (in this example I changed it from 128 to 64)

Screenshot_20190430_103308


Screenshot_20190430_102031

PGs are not yet set to 64 but the task says the pool update has been finished:
Screenshot_20190430_102213

Another thing I was wondering about: would it make sense to display something like "(Updating...)" behind the pool name when changing the PGs, similar to when a pool is being created?


Screenshot_20190430_105027

**Comment by Devp00l** [comment]
@callithea
> Now it's possible to decrease the pg's of an existing pool. The decrease
> will take place over time like the increase. As for both no notification
> will be created when this is done (issue #39482).
As described in the description to map the increase or decrease on a background task needs to be done. Therefore I created issue #39482.

**Comment by callithea** [comment]
jenkins test make check arm64

**Comment by Devp00l** [comment]
jenkins test make check arm64

**Comment by LenzGr** [comment]
> Another thing I was wondering about: would it make sense to display something like "(Updating...)" behind the pool name when changing the PGs, similar to when a pool is being created?

I actually like that idea. +1

**Comment by Devp00l** [comment]
@LenzGr that will be done in my new PR which I'm about to create ;)

**Comment by LenzGr** [comment]
Let's keep this on the radar for backporting into Nautilus, once these suggestions have also been implemented.

**Pull request merged**